[2024-08-13T00:30:10.491+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: data_pipeline_api.analyze_data scheduled__2024-08-12T00:30:00+00:00 [queued]>
[2024-08-13T00:30:10.496+0000] {taskinstance.py:1157} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: data_pipeline_api.analyze_data scheduled__2024-08-12T00:30:00+00:00 [queued]>
[2024-08-13T00:30:10.496+0000] {taskinstance.py:1359} INFO - Starting attempt 1 of 2
[2024-08-13T00:30:10.503+0000] {taskinstance.py:1380} INFO - Executing <Task(BashOperator): analyze_data> on 2024-08-12 00:30:00+00:00
[2024-08-13T00:30:10.508+0000] {standard_task_runner.py:57} INFO - Started process 36910 to run task
[2024-08-13T00:30:10.510+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'data_pipeline_api', 'analyze_data', 'scheduled__2024-08-12T00:30:00+00:00', '--job-id', '38', '--raw', '--subdir', 'DAGS_FOLDER/spark-jobs.py', '--cfg-path', '/tmp/tmpguyfhmem']
[2024-08-13T00:30:10.511+0000] {standard_task_runner.py:85} INFO - Job 38: Subtask analyze_data
[2024-08-13T00:30:10.538+0000] {task_command.py:415} INFO - Running <TaskInstance: data_pipeline_api.analyze_data scheduled__2024-08-12T00:30:00+00:00 [running]> on host 27ef58cc88fc
[2024-08-13T00:30:10.584+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='data_pipeline_api' AIRFLOW_CTX_TASK_ID='analyze_data' AIRFLOW_CTX_EXECUTION_DATE='2024-08-12T00:30:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-08-12T00:30:00+00:00'
[2024-08-13T00:30:10.586+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2024-08-13T00:30:10.587+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'bash /opt/***/jobs/spark-submit.sh /opt/***/jobs/analyze_data.py ']
[2024-08-13T00:30:10.593+0000] {subprocess.py:86} INFO - Output:
[2024-08-13T00:30:10.615+0000] {subprocess.py:93} INFO - /opt/***/jobs/analyze_data.py
[2024-08-13T00:30:10.695+0000] {subprocess.py:93} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-08-13T00:30:12.226+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-08-13T00:30:12.978+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO SparkContext: Running Spark version 3.5.1
[2024-08-13T00:30:12.978+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO SparkContext: OS info Linux, 6.6.32-linuxkit, aarch64
[2024-08-13T00:30:12.978+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO SparkContext: Java version 11.0.23
[2024-08-13T00:30:12.997+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO ResourceUtils: ==============================================================
[2024-08-13T00:30:12.997+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-08-13T00:30:12.997+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO ResourceUtils: ==============================================================
[2024-08-13T00:30:12.997+0000] {subprocess.py:93} INFO - 24/08/13 00:30:12 INFO SparkContext: Submitted application: FinanceDataAnalysis
[2024-08-13T00:30:13.015+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 1024, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-08-13T00:30:13.020+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO ResourceProfile: Limiting resource is cpus at 1 tasks per executor
[2024-08-13T00:30:13.021+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-08-13T00:30:13.061+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SecurityManager: Changing view acls to: ***
[2024-08-13T00:30:13.061+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SecurityManager: Changing modify acls to: ***
[2024-08-13T00:30:13.062+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SecurityManager: Changing view acls groups to:
[2024-08-13T00:30:13.062+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SecurityManager: Changing modify acls groups to:
[2024-08-13T00:30:13.062+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users with view permissions: ***; groups with view permissions: EMPTY; users with modify permissions: ***; groups with modify permissions: EMPTY
[2024-08-13T00:30:13.220+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO Utils: Successfully started service 'sparkDriver' on port 38461.
[2024-08-13T00:30:13.244+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SparkEnv: Registering MapOutputTracker
[2024-08-13T00:30:13.262+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SparkEnv: Registering BlockManagerMaster
[2024-08-13T00:30:13.274+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-08-13T00:30:13.274+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-08-13T00:30:13.277+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-08-13T00:30:13.289+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-0b30c91d-0086-43fa-8d44-03b97899871a
[2024-08-13T00:30:13.298+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO MemoryStore: MemoryStore started with capacity 1458.4 MiB
[2024-08-13T00:30:13.312+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-08-13T00:30:13.411+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO JettyUtils: Start Jetty 0.0.0.0:4040 for SparkUI
[2024-08-13T00:30:13.450+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2024-08-13T00:30:13.455+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO Utils: Successfully started service 'SparkUI' on port 4041.
[2024-08-13T00:30:13.478+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO SparkContext: Added JAR file:///opt/bitnami/spark/resources/elasticsearch-spark-30_2.12-8.4.3.jar at spark://27ef58cc88fc:38461/jars/elasticsearch-spark-30_2.12-8.4.3.jar with timestamp 1723509012967
[2024-08-13T00:30:13.552+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2024-08-13T00:30:13.585+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO TransportClientFactory: Successfully created connection to spark-master/172.18.0.4:7077 after 16 ms (0 ms spent in bootstraps)
[2024-08-13T00:30:13.720+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20240813003013-0028
[2024-08-13T00:30:13.734+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 42637.
[2024-08-13T00:30:13.734+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO NettyBlockTransferService: Server created on 27ef58cc88fc:42637
[2024-08-13T00:30:13.737+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-08-13T00:30:13.741+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 27ef58cc88fc, 42637, None)
[2024-08-13T00:30:13.746+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManagerMasterEndpoint: Registering block manager 27ef58cc88fc:42637 with 1458.4 MiB RAM, BlockManagerId(driver, 27ef58cc88fc, 42637, None)
[2024-08-13T00:30:13.749+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 27ef58cc88fc, 42637, None)
[2024-08-13T00:30:13.751+0000] {subprocess.py:93} INFO - 24/08/13 00:30:13 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 27ef58cc88fc, 42637, None)
[2024-08-13T00:30:14.001+0000] {subprocess.py:93} INFO - 24/08/13 00:30:14 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-08-13T00:30:14.176+0000] {subprocess.py:93} INFO - 24/08/13 00:30:14 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-08-13T00:30:14.178+0000] {subprocess.py:93} INFO - 24/08/13 00:30:14 INFO SharedState: Warehouse path is 'file:/tmp/***tmptb746y9p/spark-warehouse'.
[2024-08-13T00:30:14.886+0000] {subprocess.py:93} INFO - 24/08/13 00:30:14 INFO Version: Elasticsearch Hadoop v8.4.3 [a7873eb469]
[2024-08-13T00:30:15.765+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO StandaloneAppClient$ClientEndpoint: Master removed worker worker-20240809124721-172.18.0.6-35359: Not receiving heartbeat for 60 seconds
[2024-08-13T00:30:15.767+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO StandaloneSchedulerBackend: Worker worker-20240809124721-172.18.0.6-35359 removed: Not receiving heartbeat for 60 seconds
[2024-08-13T00:30:15.767+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO StandaloneAppClient$ClientEndpoint: Master removed worker worker-20240809124721-172.18.0.7-36933: Not receiving heartbeat for 60 seconds
[2024-08-13T00:30:15.768+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO StandaloneSchedulerBackend: Worker worker-20240809124721-172.18.0.7-36933 removed: Not receiving heartbeat for 60 seconds
[2024-08-13T00:30:15.770+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO TaskSchedulerImpl: Handle removed worker worker-20240809124721-172.18.0.6-35359: Not receiving heartbeat for 60 seconds
[2024-08-13T00:30:15.770+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO TaskSchedulerImpl: Handle removed worker worker-20240809124721-172.18.0.7-36933: Not receiving heartbeat for 60 seconds
[2024-08-13T00:30:15.776+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO DAGScheduler: Shuffle files lost for worker worker-20240809124721-172.18.0.6-35359 on host 172.18.0.6
[2024-08-13T00:30:15.780+0000] {subprocess.py:93} INFO - 24/08/13 00:30:15 INFO DAGScheduler: Shuffle files lost for worker worker-20240809124721-172.18.0.7-36933 on host 172.18.0.7
[2024-08-13T00:30:17.059+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2024-08-13T00:30:17.060+0000] {subprocess.py:93} INFO -   File "/opt/***/jobs/analyze_data.py", line 48, in <module>
[2024-08-13T00:30:17.060+0000] {subprocess.py:93} INFO -     save_analysis_result(top_gainers, "data/top_gainers.csv")
[2024-08-13T00:30:17.060+0000] {subprocess.py:93} INFO -   File "/opt/***/jobs/analyze_data.py", line 41, in save_analysis_result
[2024-08-13T00:30:17.061+0000] {subprocess.py:93} INFO -     df.coalesce(1).write.mode("overwrite").csv(file_path, header=True)
[2024-08-13T00:30:17.061+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/readwriter.py", line 1864, in csv
[2024-08-13T00:30:17.061+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2024-08-13T00:30:17.062+0000] {subprocess.py:93} INFO -   File "/home/***/.local/lib/python3.11/site-packages/pyspark/python/lib/pyspark.zip/pyspark/errors/exceptions/captured.py", line 185, in deco
[2024-08-13T00:30:17.068+0000] {subprocess.py:93} INFO - pyspark.errors.exceptions.captured.AnalysisException: [UNSUPPORTED_DATA_TYPE_FOR_DATASOURCE] The CSV datasource doesn't support the column `_metadata` of the type "MAP<STRING, STRING>".
[2024-08-13T00:30:17.114+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO SparkContext: Invoking stop() from shutdown hook
[2024-08-13T00:30:17.114+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO SparkContext: SparkContext is stopping with exitCode 0.
[2024-08-13T00:30:17.124+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO SparkUI: Stopped Spark web UI at http://27ef58cc88fc:4041
[2024-08-13T00:30:17.132+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-08-13T00:30:17.136+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO StandaloneSchedulerBackend$StandaloneDriverEndpoint: Asking each executor to shut down
[2024-08-13T00:30:17.152+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-08-13T00:30:17.162+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO MemoryStore: MemoryStore cleared
[2024-08-13T00:30:17.163+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO BlockManager: BlockManager stopped
[2024-08-13T00:30:17.167+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-08-13T00:30:17.168+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-08-13T00:30:17.176+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO SparkContext: Successfully stopped SparkContext
[2024-08-13T00:30:17.176+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO ShutdownHookManager: Shutdown hook called
[2024-08-13T00:30:17.176+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO ShutdownHookManager: Deleting directory /tmp/spark-ca461b1a-4582-4da4-840e-92ac13bb75cb/pyspark-40fa34c8-714a-4cc3-9604-905530538cd6
[2024-08-13T00:30:17.180+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO ShutdownHookManager: Deleting directory /tmp/spark-ca461b1a-4582-4da4-840e-92ac13bb75cb
[2024-08-13T00:30:17.183+0000] {subprocess.py:93} INFO - 24/08/13 00:30:17 INFO ShutdownHookManager: Deleting directory /tmp/spark-e4c82455-2701-470a-81f9-6a6858b4e413
[2024-08-13T00:30:17.217+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2024-08-13T00:30:17.234+0000] {taskinstance.py:1935} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/operators/bash.py", line 210, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2024-08-13T00:30:17.239+0000] {taskinstance.py:1398} INFO - Marking task as UP_FOR_RETRY. dag_id=data_pipeline_api, task_id=analyze_data, execution_date=20240812T003000, start_date=20240813T003010, end_date=20240813T003017
[2024-08-13T00:30:17.251+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 38 for task analyze_data (Bash command failed. The command returned a non-zero exit code 1.; 36910)
[2024-08-13T00:30:17.277+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-08-13T00:30:17.305+0000] {taskinstance.py:2776} INFO - 0 downstream tasks scheduled from follow-on schedule check
